src/Compiler/Driver/CompilerOptions.fs

```diff
@@ -1972,31 +1972,31 @@ let deprecatedFlagsFsc tcConfigB =
 // OptionBlock: Miscellaneous options
 //-----------------------------------
 
-let DisplayBannerText tcConfigB =
+let DisplayBannerText tcConfigB print =
     if tcConfigB.showBanner then
-        (printfn "%s" tcConfigB.productNameForBannerText
-         printfn "%s" (FSComp.SR.optsCopyright ()))
+        (print $"{tcConfigB.productNameForBannerText}\n"
+         print $"{FSComp.SR.optsCopyright ()}\n")
 
 /// FSC only help. (FSI has it's own help function).
-let displayHelpFsc tcConfigB (blocks: CompilerOptionBlock list) =
-    DisplayBannerText tcConfigB
-    PrintCompilerOptionBlocks blocks
-    exit 0
+let DisplayHelpFsc tcConfigB (blocks: CompilerOptionBlock list) print exit =
```

Comment:
Please add comments that you are passing the functions for unit-testing purposes.

I get why you're doing this in this case - though we don't do a lot of functional dependency injection in the compiler. Every time it's done please add a comment.

Reply:
I'm not totally keen on the names `print` and `exit` here.  It's just really easy to get these mixed up with the built-in `exit` (and possible future `print`) when refactoring code, which might inject a bad bug into the system.  So please call them `printImpl` or `fPrint` or something indicating they are a function parameter

Reply:
So I eventually decided to follow Petr's advice and remove these guys completely, instead refactoring the code in a different direction.

@0101 @vzarytovskii @dsyme